tests/utils/spark_operator.go

```diff
@@ -1,135 +1,158 @@
+	Status    string
 }
 
-func InstallSparkOperator() *SparkOperatorInstallation {
+func InstallSparkOperator() (*SparkOperatorInstallation, error) {
 	return InstallSparkOperatorWithNamespace(DefaultNamespace)
 }
 
-func InstallSparkOperatorWithNamespace(namespace string) *SparkOperatorInstallation {
+func InstallSparkOperatorWithNamespace(namespace string) (*SparkOperatorInstallation, error) {
 	clientSet, err := GetK8sClientSet()
 	if err != nil {
 		log.Fatal(err)
 		panic(err)
 	}
 
-	err = installSparkOperatorWithHelm(namespace)
-	if err != nil {
-		log.Fatal(err)
-		panic(err)
-	}
-
 	spark := SparkOperatorInstallation{
 		Namespace: namespace,
 		Clients:   clientSet,
 	}
-	return &spark
-}
+	spark.CleanUp()
 
-func (spark *SparkOperatorInstallation) CleanUp() {
-	uninstallSparkOperatorWithHelm(spark.Namespace)
+	installScript := exec.Command("bash", path.Join(TestDir, "../scripts/install_operator.sh"))
+	if strings.Contains(OperatorImage, ":") {
+		operatorImage := strings.Split(OperatorImage, ":")
+		installScript.Env = append(os.Environ(),
+			"NAMESPACE="+namespace,
+			"OPERATOR_IMAGE_NAME="+operatorImage[0],
+			"OPERATOR_VERSION="+operatorImage[1])
+	} else {
+		installScript.Env = append(os.Environ(),
+			"NAMESPACE="+namespace,
+			"OPERATOR_IMAGE_NAME="+OperatorImage)
+	}
+	out, err := installScript.CombinedOutput()
+	log.Infof("KUDO Spark operation installation script output:\n%s", out)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceNameRE, _ := regexp.Compile(`instance\.kudo\.dev/v1alpha1/(spark-\w{6}) created`)
+	parsedInstanceName := instanceNameRE.FindStringSubmatch(string(out))
+	spark.InstanceName = parsedInstanceName[1]
+
+	spark.waitForInstanceStatus("COMPLETE")
+
+	return &spark, nil
 }
 
-func (spark *SparkOperatorInstallation) OperatorPod() (coreV1.Pod, error) {
-	pods, err := spark.Clients.CoreV1().Pods(spark.Namespace).List(v1.ListOptions{LabelSelector: "app.kubernetes.io/name=sparkoperator"})
+func (spark *SparkOperatorInstallation) CleanUp() {
+	// So far multiple Spark operator instances in one namespace is not a supported configuration, so whole namespace will be cleaned
+	log.Infof("Uninstalling _all_ kudo spark operator instances and versions from %s", spark.Namespace)
+	instances, _ := getInstanceNames(spark.Namespace)
+	versions, _ := getOperatorVersions(spark.Namespace)
+
+	if instances != nil {
+		for _, instance := range instances {
+			DeleteResource(spark.Namespace, "instance.kudo.dev", instance)
+		}
+	}
 
-	if err != nil {
-		return pods.Items[0], nil
-	} else if len(pods.Items) != 1 {
-		return pods.Items[0], nil
-	} else if !strings.HasPrefix(pods.Items[0].Name, OperatorName) {
-		return pods.Items[0], nil
+	if versions != nil {
+		for _, version := range versions {
+			DeleteResource(spark.Namespace, "operatorversion.kudo.dev", version)
+		}
 	}
 
-	return pods.Items[0], nil
+	DeleteResource(spark.Namespace, "operator.kudo.dev", "spark")
+	//DropNamespace(spark.Clients, spark.Namespace)
```

Comment:
still valid or should we delete?

Reply:
`remove_operator.py` doesn't remove the namespace, but on second though I think it's valid to do so in the tests. I've un-commented the line.